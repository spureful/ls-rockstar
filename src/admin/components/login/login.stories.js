import login from './login.vue';
import "../../default.pcss"

export default {
	title: "Login"
}

export const defaultLogin = () => ({
	components: {login},
	template: `
		<login />
	`
})