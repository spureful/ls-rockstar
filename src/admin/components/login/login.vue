<template lang="pug">
.login-root
	h1.login__title Авторизация
	button(type="btn" @click="activeForm = true").login__btn-wrap.auth__btn Открыть форму авторизации
	.login__wrapper(v-if='activeForm')
		
		form.login__form.auth(@submit.prevent="toLogin")
			button(type='button' @click.prevent="activeForm = false").login__close
			h2.auth__title
			label.auth__lbl
				span.auth__input-title Логин
				input(type="text"   v-model="user.name" required).auth__input.auth__input--login
			label.auth__lbl
				span.auth__input-title Пароль
				input(type="password" v-model="user.password" ).auth__input.auth__input--password
			button(type="submit").auth__btn.auth__submit Авторизоваться

</template>


<script type="module">
import { mapActions, mapGetters } from "vuex";
import $axios from "../../requests"

	export default {
		components: {

		},
		data() {
			return {
				activeForm: true, 
				user: {
					name: '',
					password: ''
				}
			
			}
		},
		methods: {
		...mapActions( ["loginUser"]),
			async toLogin() {
				if (this.validForm()) {
					try {
						const {data: {token}} = await this.loginUser(this.user);
					
						localStorage.setItem("token", token);
					  	this.$router.replace('/about');
					} 
					catch (error) {
						console.log(error.message)
					}
				}		
			},
			validForm() {
				if (!this.user.name || !this.user.password) {
				
					return false
				} else {
					return true
				}
				
				
			}
		}
	}

</script>



<style lang="pcss" scoped>
	.login-root {
		position: absolute;
		top: 0;
		left: 0;
		right: 0;
		bottom: 0;
		padding: 50px;
		background-image: url(../../../images/content/promo/promo-bg.jpg);
		-webkit-background-size: cover;
		background-size: cover;
	}
	
	.login__title {
		color: #fff;
		margin-bottom: 20px;
	}
	
	.login__wrapper {
		position: absolute;
		top: ;
		left: 0;
		right: 0;
		bottom: 0;
		width: 100%;
		height: 100%;
		background-color: rgba(0,0,0, .6);
		display: -webkit-flex;
		display: -moz-flex;
		display: -ms-flex;
		display: -o-flex;
		display: flex;
		justify-content: center;
		-ms-align-items: center;
		align-items: center;
	}
	
	.login__form  {
		display: -webkit-flex;
		display: -moz-flex;
		display: -ms-flex;
		display: -o-flex;
		display: flex;
		-webkit-flex-direction: column;
		-moz-flex-direction: column;
		-ms-flex-direction: column;
		-o-flex-direction: column;
		flex-direction: column;
		-ms-align-items: center;
		align-items: center;
		background-color: #fff;
		padding: 30px;
	}
	
	.login__close {
		-ms-align-self: flex-end;
		align-self: flex-end;
		background: svg-load('remove.svg', fill='#000', width=100%, height=100%); 
		width: 15px;
		height: 15px;
		border: none;
	}
	
	.auth__lbl {
		display: -webkit-flex;
		display: -moz-flex;
		display: -ms-flex;
		display: -o-flex;
		display: flex;
		-webkit-flex-direction: column;
		-moz-flex-direction: column;
		-ms-flex-direction: column;
		-o-flex-direction: column;
		flex-direction: column;
		-ms-align-items: flex-start;
		align-items: flex-start;
		margin: 15px;
	}
	
	.auth__input-title {
		color: $text-color;
		margin-bottom: 10px;
		padding-left: 20px;
	}
	
	
	.auth__input {
		border: none;
		border-bottom: 1px solid  $text-color;
		padding-bottom: 10px;
		padding-left: 30px;
	}
	
	.auth__input--login {
		background: svg-load('user.svg', fill='$text-color', width=20, height=20) no-repeat; 
	}
	
	.auth__input--password {
		background: svg-load('key.svg', fill='$text-color', width=20, height=20) no-repeat; 
	}
	.auth__btn {
		background: linear-gradient(to right, #1d51dd, #3f35cb);
		padding: 10px;
		border: none;
		border-radius: 20px 5px;
		color: #fff;
		-webkit-transition: .7s;
		-o-transition: .7s;
		transition: .7s;
		text-transform: uppercase;
		font-weight: 700;
		
		&:hover {
			background: linear-gradient(to right,  #3f35cb, #1d51dd);
		}
	}
	


</style>
