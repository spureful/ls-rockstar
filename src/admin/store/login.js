const login = {
//	namespaced: true,
	//данные
	state: {
		user: {
		
		}

	},
	mutations: {

		SET_USER: (state, user) => {
			state.user = user
		},
		CLEAR_USER: state => (state.user = {})


	},

	///методы, которые описывают, что происходит с данными
	actions: {

		async loginUser({commit}, user) {
			try {
				const response = await this.$axios.post('/login', user);
				return response;
			} catch (error) {
				console.log(error)
			}
		},
		
		 logout({ commit }) {
            commit("CLEAR_USER");
//            removeToken();
            location.href = "/ls-advanced-portfolio"
        },
		



	}

	//вытаскивать нужные данные из state


	//описывают методы, которые будут изменять данные нашего приложения

}

export default login;
