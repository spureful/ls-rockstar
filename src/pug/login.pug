#login-form


script(type="text/x-template")#login
	form.login(@submit.prevent="enterLogin()")
		.login__container
			input(type="text" v-model="user.name" placeholder="login").login__input#input-login
			input(type="text" v-model="user.password" placeholder="password").login__input#input-password
			button(type="submit").login__input#input-password Войти