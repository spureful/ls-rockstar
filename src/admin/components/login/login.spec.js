import login from './login.vue';
import {shallowMount , createLocalVue} from "@vue/test-utils";
import Vuex from 'vuex';

const localVue = createLocalVue();
localVue.use(Vuex);


describe('На странице есть кнопка Авторизоваться', () => {
    it ('Ищем кнопку авторизации по классу', ()=> {
        
        const wrapper = shallowMount(login);
        const button = wrapper.find(".auth__submit");

        expect(button.element).toMatchSnapshot();

    });

    it ('Ищем кнопку авторизации по слову Авторизоваться', ()=> {
        
        const wrapper = shallowMount(login);
        const button = wrapper.find("button[type='submit']");

        expect(button.text()).toBe("Авторизоваться");

    });
});

describe('при клике на кнопку авторизации появляется формa', () => {
    it ('open form', async () => {
        const wrapper = shallowMount(login);
        const openBtn = wrapper.find('.login__btn-wrap');
        const form = wrapper.find('.login__form');

        openBtn.trigger('click');
        await wrapper.vm.$nextTick();
        expect(form.element).toMatchSnapshot();

    })
})


describe('форма имеет все необходимые поля', ()=> {
    it('есть поле логин', () => {
        const wrapper = shallowMount(login);
        const inputLogin = wrapper.find('.auth__input--login');

        expect(inputLogin.element).toMatchSnapshot();
    });

    it('есть поле пароль', () => {
        const wrapper = shallowMount(login);
        const inputPassword = wrapper.find('.auth__input--password');

        expect(inputPassword.element).toMatchSnapshot();
    });
});

describe('кнопка “Отправить” заблокирована до тех пор, пока не введены все необходимые данные', ()=> {
    let actions;
    let store;

    beforeEach(() => {
        actions = {
            loginUser: jest.fn()
         
        }
        store = new Vuex.Store({
          actions
        })
      })

      it ('форма не сабмитится при незаполненных полях', () => {
        const wrapper = shallowMount(login , {store, localVue});
        const form = wrapper.find('.login__form');
        const inputLogin = wrapper.find('.auth__input--login');
        const inputPassword = wrapper.find('.auth__input--password');

        inputLogin.setValue('');
        inputPassword.setValue('');
        form.trigger('submit');

        expect(actions.loginUser).not.toHaveBeenCalled();
      })
})



