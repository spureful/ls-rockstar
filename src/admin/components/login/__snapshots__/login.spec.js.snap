// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`На странице есть кнопка Авторизоваться Ищем кнопку авторизации по классу 1`] = `
<button
  class="auth__btn auth__submit"
  type="submit"
>
  Авторизоваться
</button>
`;

exports[`при клике на кнопку авторизации появляется формa open form 1`] = `
<form
  class="login__form auth"
>
  <button
    class="login__close"
    type="button"
  />
  <h2
    class="auth__title"
  />
  <label
    class="auth__lbl"
  >
    <span
      class="auth__input-title"
    >
      Логин
    </span>
    <input
      class="auth__input auth__input--login"
      required="required"
      type="text"
    />
  </label>
  <label
    class="auth__lbl"
  >
    <span
      class="auth__input-title"
    >
      Пароль
    </span>
    <input
      class="auth__input auth__input--password"
      type="password"
    />
  </label>
  <button
    class="auth__btn auth__submit"
    type="submit"
  >
    Авторизоваться
  </button>
</form>
`;

exports[`форма имеет все необходимые поля есть поле логин 1`] = `
<input
  class="auth__input auth__input--login"
  required="required"
  type="text"
/>
`;

exports[`форма имеет все необходимые поля есть поле пароль 1`] = `
<input
  class="auth__input auth__input--password"
  type="password"
/>
`;
